src/register.rs

```diff
@@ -6,13 +6,143 @@ pub mod common {
+        }
+
+        pub fn to_u8(self) -> u8 {
+            match self {
+                Mode::Reset => 0b10000000,
+                Mode::Mode(mode) => mode.to_u8(),
+            }
+        }
+    }
+
+    /// A Retry Time-value
+    ///
+    /// RTR (Retry Time-value Register) [R/W] [0x0019 – 0x001A] [0x07D0]
+    ///
+    /// From datasheet:
+    ///
+    /// RTR configures the retransmission timeout period. The unit of timeout period is
+    /// 100us and the default of RTR is ‘0x07D0’ or ‘2000’. And so the default timeout period
+    /// is 200ms(100us X 2000).
+    /// During the time configured by RTR, W5500 waits for the peer response to the packet
+    /// that is transmitted by Sn_CR(CONNECT, DISCON, CLOSE, SEND, SEND_MAC, SEND_KEEP
+    /// command). If the peer does not respond within the RTR time, W5500 retransmits the
+    /// packet or issues timeout.
+    ///
+    ///
+    /// > Ex) When timeout-period is set as 400ms, RTR = (400ms / 1ms) X 10 = 4000(0x0FA0)
+    #[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub struct RetryTime(pub(crate) u16);
+
+    impl RetryTime {
+        #[inline]
+        pub fn to_u16(&self) -> u16 {
+            self.0
+        }
+
+        #[inline]
+        pub fn to_register(&self) -> [u8; 2] {
+            self.0.to_be_bytes()
+        }
+
+        #[inline]
+        pub fn from_register(register: [u8; 2]) -> Self {
+            Self(u16::from_be_bytes(register))
+        }
+
+        #[inline]
+        pub fn from_millis(milliseconds: u16) -> Self {
+            Self(milliseconds * 10)
+        }
+
+        #[inline]
+        pub fn to_millis(&self) -> u16 {
+            self.0 / 10
+        }
+    }
+
+    impl Default for RetryTime {
+        fn default() -> Self {
+            Self::from_millis(200)
+        }
+    }
+
+    /// RCR (Retry Count Register) [R/W] [0x001B] [0x08]
+    ///
+    /// For more details check out the rest of the datasheet documentation on the Retry count.
+    ///
+    /// From datasheet:
+    ///
+    /// RCR configures the number of time of retransmission. When retransmission occurs
+    /// as many as ‘RCR+1’, Timeout interrupt is issued (Sn_IR[TIMEOUT] = ‘1’).
+    ///
+    /// > Ex) RCR = 0x0007
+    ///
+    /// The timeout of W5500 can be configurable with RTR and RCR. W5500 has two kind
+    /// timeout such as Address Resolution Protocol (ARP) and TCP retransmission.
+    ///
+    #[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub struct RetryCount(pub u8);
+
+    impl RetryCount {
+        #[inline]
+        pub fn to_u8(&self) -> u8 {
+            self.0
+        }
+
+        #[inline]
+        pub fn to_register(&self) -> [u8; 1] {
+            self.0.to_be_bytes()
+        }
+
+        #[inline]
+        pub fn from_register(register: [u8; 1]) -> Self {
+            Self(register[0])
+        }
+    }
```

Comment:
I'm not sure that wrapping these in custom types makes them easier to use. Why not just take a simple arg from the user as milliseconds and handle conversion directly in the usage function?

Reply:
I see them as a wrappers of the raw Register values themselves. Makes it much more simple to follow the datasheet for example and see how this register is represented in the device itself.

Logging values makes it much more clear too.